forms-flow-api/src/formsflow_api/resources/form_process_mapper.py

```diff
@@ -400,16 +400,33 @@ def get(form_id: string, submission_id: string):
         """Form rendering method."""
         formio_service = FormioService()
         form_io_url = current_app.config.get("FORMIO_URL")
+        is_form_adapter = current_app.config.get("CUSTOM_SUBMISSION_ENABLED")
         form_io_token = formio_service.get_formio_access_token()
-        form_url = form_io_url + "/form/" + form_id + "/submission/" + submission_id
+
+        if is_form_adapter:
```

Comment:
Good for now, but we need to extract these logic to python module and re-use it!
I will create a tech debt ticket

Reply:
Yes. Previously form detail and submission data were both fetched on the javascript side by the FormIO library. Now if form_adapater enabled, submission data fetch logic we do separate API calls, but I kept the same in javascript only.
As you said it's recommended to have these submission data fetching logic from the form adapter on the python side and can be re-used in similar scenarios.